src/main/java/org/prebid/server/auction/model/PrebidLog.java

```diff
@@ -0,0 +1,42 @@
+package org.prebid.server.auction.model;
+
+import lombok.Value;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+@Value(staticConstructor = "of")
+public class PrebidLog {
+
+    Map<String, Set<PrebidMessage>> log = new HashMap<>();
+
+    public void logMessage(PrebidMessage prebidMessage) {
+        prebidMessage.getTags().forEach(
+                tag -> log.computeIfAbsent(tag, type -> new HashSet<>()).add(prebidMessage));
+    }
+
+    public Set<PrebidMessage> getPrebidMessagesByTag(String tag) {
+        final Set<PrebidMessage> prebidMessages = log.get(tag);
+        return prebidMessages == null ? Collections.emptySet() : prebidMessages;
+    }
+
+    public Set<PrebidMessage> getPrebidMessagesByTags(List<String> tags) {
+        Set<PrebidMessage> result = new HashSet<>();
+        tags.stream()
+                .map(log::get)
+                .filter(Objects::nonNull)
+                .forEach(result::addAll);
+        return result;
```

Comment:
I guess name of the method implies finding all messages by set of tags. So, lets rename this method to `findMessagesByTags`.
Also, please, add method for filtering by tags, lets call it `filterMessagesByTags`.
for example, lets assume we have these messages:
`Message_1(["bidder", "error"], "Some ugly bidder error"), Message_2(["bidder", "warning"], "Just a warning")`

For a call like `prebidLog.filterMessagesByTags(List.of("bidder", "warning"))` it will return you only Message_2.